src/validate.ts

```diff
@@ -0,0 +1,58 @@
+const crypto = require('crypto');
+
+const core = require('@actions/core');
+
+const request = require('requestretry');
+
+const validateUploader = async (body) => {
+  const version = getVersion(body);
+  if (version === null) {
+    core.warning('Codecov could not identify the bash uploader version.');
+    return false;
+  }
+
+  for (const i of [1, 256, 512]) {
```

Comment:
What's the purpose of checking all the checksums? Is this intended?

Reply:
this was intentional, although it is unlikely that one would match and another wouldn't, I'd prefer to be thorough